olclient/entity_helpers/work.py

```diff
@@ -138,6 +139,19 @@ def rm_subjects(self, subjects, comment=''):
             data['subjects'] = list(set(data['subjects']) - set(subjects))
             return self.OL.session.put(url, json.dumps(data))
 
+        def _delete_editions(self, comment: str) -> None:
+            edition_olids: List[str] = [edition_data.olid for edition_data in self.editions]
+            if len(edition_olids) > 0:
+                editions_cleanup_response: Response = self.OL.delete_many(edition_olids, comment)
+                if editions_cleanup_response.ok is False:
+                    raise Exception(f'Could not delete all editions of work {self.olid}. '
+                                    f'response: {editions_cleanup_response}')
+
+        def delete(self, comment: str, editions: bool = False):
```

Comment:
Actually, on second thought, I think we might want this to be `True` by default, and we should display/ask for confirmation. Something like this:

```
>>> ol.Work.get('/works/OL1W').delete('Remove t-shirt book')
Delete https://openlibrary.org/works/OL123W and its 16 editions? (y/n)
```

We can then add an option `confirm = True`, which will hide the confirm message if a bulk deletion is necessary. But I think by default these is kind of like an OL cli, so we should show a confirmation.